### PR TITLE
Modify workflow to use PyPI

### PR DESCRIPTION
**Pull request by TheMichaelHu**
I registered this project under `ml-pipeline-gen` in PyPI, so it can now be installed in pip.
- created a top level `setup.py` and removed `requirements.py`
- changed the workflow so user now installs the package, copies an example, and runs the demo
- manual tested the 4 original examples (tf, sklearn, xgboost, kfp), moved other examples to `experimental` dir for now

### CODE REVIEW

**Review by smhosein — COMMENTED** [review]
This looks great Michael, thanks for adding this to PyPI. I saw you moved taxi examples to experimental, what were the breaking changes?

**Comment by TheMichaelHu** [comment]
The breaking change was moving from using `ml-pipeline-gen` as a local package to turning it into a package that can be imported from anywhere. There are a lot of assumptions we made that had to change. Actually, it might make sense if I just fix the taxi examples in this PR. I was just worried it would get too big, but it's already huge so 🤷‍♂️

**Comment by TheMichaelHu** [comment]
@smhosein oh right, another reason I moved it to experimental was because we're missing the `config.yaml.example` files for these examples. The `.gitignore` actually ignores all `config.yaml`s, and each example has a `config.yaml.example` instead. If you send me those configs, I'll be able to fix the rest.

**Comment by TheMichaelHu** [comment]
btw it might make sense to break out the documentation improvement into a separate PR if the k8 stuff becomes too big of a blocker

File: ml_pipeline_gen/models.py [hunk 1]
```diff
@@ -161,6 +163,13 @@ def _write_template(self, env, template_path, args, dest):
         with open(dest, "w+") as f:
             f.write(body)
 
+    def _write_static(self):
+        """Copies static files to the working directory."""
+        root_dir = pathlib.Path(__file__).parent.resolve()
+        for d in root_dir.joinpath("static").iterdir():
```
Comment (ashok133) [inline ml_pipeline_gen/models.py:169]: This throws a `FileNotFoundError` for me. According to readme instructions, my current dir is mounted on a folder `sklearn-demo/` and it's trying to copy files from `ml-pipeline-gen/static/` which doesn't exist (looked in my venv's ml-pipeline-gen package installed from PyPI).

Comment (TheMichaelHu) [reply ml_pipeline_gen/models.py:169]: oh shoot, I might need to do something with manifests

**Comment by ashok133** [comment]
For the KFP demo, I think we may have to move `component_lib.py` and `component_spec.yaml` out of `experimental/` for the package since `component_lib.generate_component()` is being used in `pipelines.py` but being imported as `from ml_pipeline_gen.component_lib import generate_component` throwing a `ModuleNotFoundError`. 

Rest LGTM 👍. As @smhosein said, thanks for doing this!

**Comment by ashok133** [comment]
> For the KFP demo, I think we may have to move `component_lib.py` and `component_spec.yaml` out of `experimental/` for the package since `component_lib.generate_component()` is being used in `pipelines.py` but being imported as `from ml_pipeline_gen.component_lib import generate_component` throwing a `ModuleNotFoundError`.
> 
> Rest LGTM 👍. As @smhosein said, thanks for doing this!

Or include `experimental/` as a part of the PyPI distribution and change the import? Either should work I think.

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1,5 @@
+graft ml_pipeline_gen/templates
+graft ml_pipeline_gen/static
+
+prune **/experimental
```
Comment (ashok133) [inline MANIFEST.in:4]: `graft **/experimental` for the time being, or if we're confident, pull out `component_lib.py` and `component_spec.yaml`. Please correct me if I'm mistaken here.

**Comment by smhosein** [comment]
@TheMichaelHu I sent the config examples.

File: MANIFEST.in (hunk 2 above)
Comment (TheMichaelHu) [reply MANIFEST.in:4]: I'm ignoring experimental to reduce the size of the source distribution. Is there anything in experimental that we currently recommend users use?

Comment (smhosein) [reply MANIFEST.in:4]: I think keeping the distribution size small is good, for the CAIP, we do not need anything in experimental.

**Review by smhosein — APPROVED** [review]
This is a hefty but needed change, thanks again Michael. On the CAIP side, everything LGTM.

**Pull request merged**